main/webapp/modules/core/scripts/reconciliation/standard-service-panel.js

```diff
@@ -306,32 +349,34 @@ ReconStandardServicePanel.prototype.start = function() {
   }
 
   var columnDetails = [];
+  var invalidState = false;
   $.each(
     this._panel.find('input[name="property"]'),
     function(index) {
-      var property = $(this).data("data.suggest");
-      if (property && property.id && include[index].checked) {
+      let property = $(this).data("data.suggest");
+      let hasSuggest = property && property.id && property.name;
+      if (hasSuggest) {
         columnDetails.push({
-          column: this.getAttribute("columnName"),
+          column: $(this).data("columnName"),
           property: {
             id: property.id,
             name: property.name
           }
         });
       } else {
-        var property = jQueryTrim(this.value);
-        if (property && include[index].checked) {
-          columnDetails.push({
-            column: this.getAttribute("columnName"),
-            property: {
-              id: property,
-              name: property
-            }
-          });
+        let value = jQueryTrim(this.value);
+        let hasValue = value && value.length > 0;
+        if (hasValue) {
+          alert('Column '+$(this).data("columnName")+' is not mapped.');
```

Comment:
I would localize this message with `$.i18n` and I suggest the column name gets enclosed by double quotes to make it stand out more and separate it from our own text.
Also, if multiple columns are left in an unmapped state but non-empty text field, a alert will be generated for each of them. Having sequences of alert like this is not amazing UX, so I would suggest to either only show the error for a single column (the first one the validation code encounters). Adding all the column names in a single alert could also be an option, but it's probably harder to implement and localize.

Reply:
I Agree, but I would like to keep it like this for now because there is a different error mechanism being used in the Wikibase schema editor where invalid fields are highlighted in red, so I may change it to that later.  The window.alert() is a placeholder for now.